homeassistant/components/auth/login_flow.py

```diff
@@ -146,12 +148,66 @@ async def get(self, request: web.Request) -> web.Response:
                 message_code="onboarding_required",
             )
 
-        return self.json(
-            [
-                {"name": provider.name, "id": provider.id, "type": provider.type}
-                for provider in hass.auth.auth_providers
-            ]
-        )
+        try:
+            remote_address = ip_address(request.remote)  # type: ignore[arg-type]
+        except ValueError:
+            return self.json_message(
+                message="Invalid remote IP",
+                status_code=HTTPStatus.BAD_REQUEST,
+                message_code="invalid_remote_ip",
+            )
+
+        cloud_connection = is_cloud_connection(hass)
+
+        providers = []
+        additonal_data = {}
+        for provider in hass.auth.auth_providers:
+            if provider.type == "trusted_networks":
+                if cloud_connection:
+                    # Skip quickly as trusted networks are not available on cloud
+                    continue
+
+                from homeassistant.auth.providers.trusted_networks import (  # pylint: disable=import-outside-toplevel
+                    TrustedNetworksAuthProvider,
+                )
```

Comment:
If it's only used for typing we can impptt it under a `TYPE_CHECKING` check at the top.